File: leetcode/2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array-unsolved.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        ways: int = 0
        size: int = len(nums)
        left_sum: List[int] = [0] * size
        right_sum: List[int] = [0] * size
        all_sum: int = sum(nums)
        for i in range(size - 1):
            left_sum[i] = nums[i] if i == 0 else nums[i] + left_sum[i - 1]
            right_sum[i] = all_sum - left_sum[i]
            if left_sum[i] >= right_sum[i]:
                ways += 1
        return ways

File: leetcode/2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array-unsolved.cpp
class Solution {
public:
    int32_t waysToSplitArray(const std::vector<int32_t>& nums) {
        const int32_t size = static_cast<int32_t>(nums.size());
        if (size <= 1) { return 0; }

        std::vector<intmax_t> left_sum;
        std::vector<intmax_t> right_sum;
        left_sum.resize(size - 1);
        right_sum.resize(size - 1);
        
        intmax_t sum = std::accumulate(nums.begin(), nums.end(), 0);
        //printf("sum=%i\n", sum);
        int32_t split_points_cnt = 0;
        for (int32_t i = 0; i < size - 1; ++i) {
            left_sum[i] = i == 0 ? nums[i] : nums[i] + left_sum[i - 1];
            right_sum[i] = sum - left_sum[i];
            //printf("left=%i\n", left_sum[i]);
            //printf("right-%i\n", right_sum[i]);
            if (left_sum[i] >= right_sum[i]) { split_points_cnt++; }
        }
        return split_points_cnt;
    }
};
